### DISCUSSION TITLE
How to correctly retrieve the token from the configured broker?

### DISCUSSION THREAD

**Discussion by matheushent** [keycloak/keycloak#23673] [Q&A]
Hi all.

I have been trying to retrieve a stored token from my configured broker after successful login as stated on the [docs](https://www.keycloak.org/docs/latest/server_admin/#retrieving-external-idp-tokens). As we all can see in the screenshot below (taken from jwt.io), my user has the `read-token` role set.

![image](https://github.com/keycloak/keycloak/assets/45343359/b531cfc4-eab2-4bf5-8824-5b95f626951c)

Then, I'm hitting `/realms/{realm}/broker/github/token` as stated on the docs but I just get 401, and of course, I'm passing the access token from Keycloak as the authentication token for this request.

![image](https://github.com/keycloak/keycloak/assets/45343359/476a71e6-0b5c-4780-b177-affb5df68502)

The curl command is similar to
```bash
curl --silent --location 'https://example.com/admin/realms/myrealm/broker/github/token' \
--header 'Authorization: Bearer $TOKEN' 
```

Any help to get this thing figured out?

**Comment by justin-tay** [comment]
Is the `/admin` in the path intended?
